### PR TITLE
Rails react v1

### PR DESCRIPTION
**Pull request by iamsjunaid**
## In this milestone of the project, I completed the following requirements 

- [x] Create a new Rails app called 'hello-rails-react'.
- [x]  Verify that the jsbundling-rails gem is added to your Gemfile.
- [x] Initialize your project with Git.
- [x] Install needed node dependencies as described in the previous lesson's tutorial.
- [x] Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- [x] Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- [x] Make sure that your project has a Postgres database set up.
- [x] Create a table for storing your messages and create 5 different greetings.
- [x] Create an API endpoint that selects a random greeting from your table
- [x] Create a static view that will be the root of your app.
- [x] Create the App component with react-router and render it in your static view.
- [x] Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- [x] Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- [x] Display the random greeting in your Greeting component.

### CODE REVIEW

File: app/javascript/components/Greetings.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// app/javascript/components/Greeting.js
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import fetchGreeting from '../redux/thunk';
```
Comment (bienvenuushindi) [inline app/javascript/components/Greetings.js:4]: :red_circle: [REQUIRED] It seems that the file `"../redux/thunk"` is currently empty, and there is no function called` "fetchGreeting"` in that file. Please ensure that the file contains the necessary code, including the definition of the `"fetchGreeting"` function, before attempting to import and use it in your code.

File: app/javascript/components/Greetings.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// app/javascript/components/Greeting.js
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
```
Comment (bienvenuushindi) [inline app/javascript/components/Greetings.js:3]: :red_circle: [REQUIRED] It appears that you are trying to import from a dependency  `'react-redux' `that doesn't exist:
 It seems that this package is either not installed or incorrectly configured in your project. Please make sure that you have 'react-redux' installed as a dependency in your project and that your project's configuration is set up correctly to use it.

File: app/javascript/application.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// Entry point for the build script in your package.json
+import "@hotwired/turbo-rails"
+import "./controllers"
+import React from 'react';
+import { Provider } from 'react-redux';
+import ReactDOM from 'react-dom';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
```
Comment (bienvenuushindi) [inline app/javascript/application.js:7]: :red_circle: [REQUIRED] It appears that you are trying to import components from 'react-router-dom,' but this dependency doesn't seem to be installed in your project. Please make sure you have installed 'react-router-dom' as a dependency using npm or yarn before attempting to import components from it.

File: app/javascript/redux/reducers/greetingSlice.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit';
```
Comment (bienvenuushindi) [inline app/javascript/redux/reducers/greetingSlice.js:1]: :red_circle: :red_circle: [REQUIRED] It appears that you are trying to import components from` @reduxjs/toolkit `but this dependency doesn't seem to be installed in your project. Please make sure you have installed` '@reduxjs/toolkit' `as a dependency using npm or yarn before attempting to import components from it.

File: app/javascript/redux/reducers/greetingSlice.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import fetchGreeting from '../thunk';
```
Comment (bienvenuushindi) [inline app/javascript/redux/reducers/greetingSlice.js:2]: :red_circle: [REQUIRED] It seems that the file `"../thunk"` is currently empty, and there is no function called` "fetchGreeting"` in that file. Please ensure that the file contains the necessary code, including the definition of the `"fetchGreeting"` function, before attempting to import and use it in your code.

File: app/javascript/application.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import store from './redux/store';
```
Comment (bienvenuushindi) [inline app/javascript/application.js:8]: :red_circle: [REQUIRED] It seems that the file `"../redux/store"` is currently empty, and there is nothing called` "store"` in that file. Please ensure that the file contains the necessary code, including the definition of the `"store"` object, before attempting to import and use it in your code.

**Review by bienvenuushindi — APPROVED** [review]
## Hi @iamsjunaid ,

Good job so far!

## Highlights!
- GitFlow ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review and here 👇._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/greetings_controller.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class Api::GreetingsController < ApplicationController
+  def random
+    random_greeting = Message.order('RANDOM()').first
```
Comment (FrederickMih) [inline app/controllers/api/greetings_controller.rb:3]: - Two apps connect together successfully and random greetings are display :rocket: 
Screenshot from 2023-09-26 10-28-48

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @iamsjunaid, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- Awesome job resolving dependencies issues raised by reviewer :+1: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**